lte/cloud/go/services/subscriberdb/storage/digest_lookup.go

```diff
@@ -0,0 +1,207 @@
+			filters = append(filters, squirrel.Lt{digestLookupLastUpdatedTimeCol: earliestUpdateTime})
+		}
+
+		rows, err := l.builder.
+			Select(digestLookupNidCol, digestLookupDigestCol, digestLookupLastUpdatedTimeCol).
+			From(digestLookupTableName).
+			Where(filters).
+			RunWith(tx).
+			Query()
+		if err != nil {
+			return nil, errors.Wrapf(err, "gets digest for networkIDs %+v", networkIDs)
+		}
+		defer sqorc.CloseRowsLogOnError(rows, "GetDigest")
+
+		digests := []DigestInfo{}
+		for rows.Next() {
+			networkID, digest, lastUpdatedTime := "", "", sql.NullInt64{}
+			err = rows.Scan(&networkID, &digest, &lastUpdatedTime)
+			if err != nil || !lastUpdatedTime.Valid {
+				return nil, errors.Wrap(err, "select digests for networkIDs, SQL row scan error")
+			}
+			digests = append(digests, DigestInfo{
+				networkID:       networkID,
+				digest:          digest,
+				lastUpdatedTime: lastUpdatedTime.Int64,
+			})
+		}
+		err = rows.Err()
+		if err != nil {
+			return nil, errors.Wrap(err, "select digests for networkID, SQL rows error")
+		}
+		return digests, nil
+	}
+
+	txRet, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	if err != nil {
+		return nil, err
+	}
+	ret := txRet.([]DigestInfo)
+	return ret, nil
+}
+
+func (l *digestLookup) SetDigest(networkID string, digest string) error {
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		sc := squirrel.NewStmtCache(tx)
+		defer sqorc.ClearStatementCacheLogOnError(sc, "SetDigest")
+
+		curUnixTime := clock.Now().Unix()
+		_, err := l.builder.
+			Insert(digestLookupTableName).
+			Columns(digestLookupNidCol, digestLookupDigestCol, digestLookupLastUpdatedTimeCol).
+			Values(networkID, digest, curUnixTime).
+			OnConflict(
+				[]sqorc.UpsertValue{
+					{Column: digestLookupDigestCol, Value: digest},
+					{Column: digestLookupLastUpdatedTimeCol, Value: curUnixTime},
+				},
+				digestLookupNidCol,
+			).
+			RunWith(sc).
+			Exec()
+		if err != nil {
+			return nil, errors.Wrapf(err, "insert digest for network %+v", networkID)
+		}
+		return nil, nil
+	}
+
+	_, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	return err
+}
+
+func (l *digestLookup) DeleteDigests(networkIDs []string) error {
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		_, err := l.builder.
+			Delete(digestLookupTableName).
+			Where(squirrel.Eq{digestLookupNidCol: networkIDs}).
+			RunWith(tx).
+			Exec()
+		if err != nil {
+			return nil, errors.Wrapf(err, "delete digests")
+		}
+		return nil, nil
+	}
+
+	_, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	return err
+}
+
+func GetDigest(l DigestLookup, networkID string) (string, int64, error) {
+	digestInfos, err := l.GetDigests([]string{networkID}, -1)
+	// There should be at most 1 row for each network ID.
+	if err != nil || len(digestInfos) == 0 {
+		return "", 0, err
+	}
+	digestInfo := digestInfos[0]
+	return digestInfo.digest, digestInfo.lastUpdatedTime, nil
+}
+
+func GetOutdatedNetworkIDs(l DigestLookup, updateDeadlineTime int64) ([]string, error) {
```

Comment:
This default value is specified by `updateIntervalSecs` in `subscriberdb_cache`'s yaml configs, since the function is only invoked in the `worker.go` code, and the caller passes down the `updateDeadlineTime` according to the caller requirements. (If a default value is defined here, it wouldn't be invoked) @hcgatewood